src/addToClass.ts

```diff
@@ -49,9 +53,10 @@ export default function addToClass(node: t.ClassDeclaration, state: ConvertState
   if (propTypes) {
     propTypes.value = mergePropTypes(propTypes.value, propTypesList, state);
   } else {
+    const isVariable = addVariables && typeNames.length === 1;
```

Comment:
I think instead of assuming that typeName === a variable, you should check for imports and validate that it's actually a variable.

Reply:
Good.

Reply:
What do you mean?

Reply:
For this?
```js
export default class ClassStandard extends React.Component<{ name: string }> {
  render() {
    return null;
  }
}
```
Like that?
```js
const isVariable = (
  state.options.declarePropTypeVariables &&
  typeNames.length === 1 &&
  state.componentTypes[typeNames[0]]
);
```

Reply:
Or do you mean to check if such a variable exists?
But it's redundant. We create it in the same location where you save this type of state:
https://github.com/milesj/babel-plugin-typescript-to-proptypes/blob/afb36e2c0260377fee8e5997611c629ed12e3ba5/src/index.ts#L210-L218

Reply:
Or do you think I'm adding a variable just for export?
That's not so. A variable is created for all types:
https://github.com/milesj/babel-plugin-typescript-to-proptypes/blob/afb36e2c0260377fee8e5997611c629ed12e3ba5/tests/__snapshots__/index.test.ts.snap#L186-L199